franklin/builder/models.py

```diff
@@ -114,16 +102,64 @@ class Build(models.Model):
     :param site: Reference to the project for this build instance
     :param git_hash: If a branch build, the git hash of the deployed code
     :param created: Date this code was built
+    :param deployed: Date this code was last deployed to an environment
     :param path: The path of the site on the static server
     """
 
+    NEW = 'NEW'
+    BUILDING = 'BLD'
+    SUCCESS = 'SUC'
+    FAILED = 'FAL'
+    STATUS_CHOICES = (
+        (NEW, _('new')),
+        (BUILDING, _('building')),
+        (SUCCESS, _('success')),
+        (FAILED, _('failed'))
+    )
+
     site = models.ForeignKey(Site, related_name='builds')
     created = models.DateTimeField(auto_now_add=True, editable=False)
     path = models.CharField(max_length=100, blank=True)
+    status = models.CharField(max_length=3, choices=STATUS_CHOICES,
+                              default=NEW)
 
     def get_path(self, site, name):
         return "{0}/{1}".format(site, name)
 
+    def can_build(self):
+        return self.status is not self.BUILDING
+
+    def deploy(self, environment):
+        if self.can_build():
+            callback = os.environ['API_BASE_URL'] + \
+                reverse('webhook:builder', args=[self.git_hash, ])
+
+            url = os.environ['BUILDER_URL'] + '/build'
+            headers = {'content-type': 'application/json'}
+            body = {
+                "deploy_key": self.site.deploy_key_secret,
+                "branch": self.branch,
+                "git_hash": self.git_hash,
+                "repo_owner": self.site.owner.name,
+                "path": self.path,
+                "repo_name": self.site.name,
+                "environment": environment.name.lower(),
+                'callback': callback
+            }
+            response = make_rest_post_call(url, headers, body)
+            if not status.is_success(response.status_code):
+                logger.error("Negative response from Builder: %s",
+                             response.status_code)
+                self.status = self.FAILED
+            else:
+                self.status = self.BUILDING
+            self.save()
+        else:
+            logger.error("Build being/been built by builder...")
+
+    def __str__(self):
+        return '%s - %s' % (self.status, self.created)
```

Comment:
maintaining status and executing the deploy call to builder......formerly the job of the Environment model, is now owned by Build model.